uaclient/files/notices.py

```diff
@@ -0,0 +1,206 @@
+import os
+from collections import namedtuple
+from enum import Enum
+from typing import Any, Dict, List, Optional  # noqa: F401
+
+from uaclient import defaults, event_logger, exceptions, system
+from uaclient.files import UAFile
+
+event = event_logger.get_event_logger()
+NoticeFileDetails = namedtuple(
+    "NoticeFileDetails", ["order_id", "label", "is_permanent"]
+)
+
+
+class Singleton(type):
+    _instances = {}  # type: Dict[Any, Any]
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(
+                *args, **kwargs
+            )
+        return cls._instances[cls]
```

Comment:
While very cool, I think the metaclass is a bit overkill here to enforce the singleton.
We can just only create one.
```python
_notices = None
def get_notices():
    if _notices is None:
        _notices = Notices()
    return _notices


def add():
    notices = get_notices()
    notices.add()
```

Reply:
I could be convinced either way :upside_down_face: maybe we should convert other places to use the Singleton metaclass

Reply:
I would rather use the simpler approach, but I don't have a strong opinion here

Reply:
FFR: we talked out of band and decided that for now we would stick with the approach we use in other places, and there will be a future task to migrate us to use the singleton metaclass everywhere